src/wireviz/DataClasses.py

```diff
@@ -50,11 +51,15 @@ def __post_init__(self):
             if len(self.pinlabels) != len(self.pins):
                 raise Exception('Given pins and pinlabels size mismatch')
 
+        # TODO: Check pincolor length
+
         # create default lists for pins (sequential) and pinlabels (blank) if not specified
         if not self.pins:
             self.pins = list(range(1, self.pincount + 1))
         if not self.pinlabels:
             self.pinlabels = [''] * self.pincount
+        if not self.pincolors:
+            self.pincolors = [None] * self.pincount
```

Comment:
Will use
```
        if self.pincolors:
            self.pincolors.extend(["--"] * (len(self.pins) - len(self.pincolors)))
```
It's shorter, appends nothing if list lengths match, and uses `pins` instead of `pincount` since `pincount` may artificially be set higher.